### PR TITLE
[Improvement][script] Add installPath verification for install.sh

### LINKED ISSUE #12522: [Improvement][script] Add installPath verification for install.sh
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When installing dolphinscheduler, execute script/install.sh
If the installPath in script/env/install_env.sh is the same as the directory where the current script executes script/install.sh, the files in the installation directory will be deleted.
Although the comments of installPath remind users that the two paths are not consistent, there are still many novice users who make mistakes and cause important files in the installation directory to be deleted
I think a simple check of installPath should be done in script/env/install_env.sh

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by muggleChen**
## Purpose of the pull request

Added verification of installPath.
Prevent some important files from being deleted because the installPath is consistent with the current script directory.
close #12522

### CODE REVIEW

**Comment by DarkAssassinator** [comment]
hi @muggleChen, please rebase upstream and fix failed CI.

File: script/install.sh [hunk 1]
```diff
@@ -27,7 +27,10 @@ echo "1.create directory"
 # If install Path equal to "/" or related path is "/" or is empty, will cause directory "/bin" be overwrite or file adding,
 # so we should check its value. Here use command `realpath` to get the related path, and it will skip if your shell env
 # without command `realpath`.
-if [ ! -d $installPath ];then
+if [ `cd ${workDir}/..;pwd` = $installPath ]; then
```
Comment (DarkAssassinator) [inline script/install.sh]: It is recommended not to use commands in condition

**Comment by muggleChen** [comment]
> hi @muggleChen, please rebase upstream and fix failed CI.

hi @DarkAssassinator,I have rebase upstream, but the CI failed how should I fix it, can you give some tips?
[ The log shows that there are files that don't exist, but I didn't delete any files](https://github.com/apache/dolphinscheduler/actions/runs/3319214527/jobs/5484892821)

**Comment by DarkAssassinator** [comment]
> > hi @muggleChen, please rebase upstream and fix failed CI.
> 
> hi @DarkAssassinator,I have rebase upstream, but the CI failed how should I fix it, can you give some tips? [ The log shows that there are files that don't exist, but I didn't delete any files](https://github.com/apache/dolphinscheduler/actions/runs/3319214527/jobs/5484892821)

because u should rebase the upsteam and `git push -f`

**Comment by muggleChen** [comment]
@DarkAssassinator  This is my first pullrequest and I think I've rebase upstream, can you help me re-run CI ? thanks :)

**Comment by EricGao888** [comment]
> @DarkAssassinator This is my first pullrequest and I think I've rebase upstream, can you help me re-run CI ? thanks :)

done

File: script/install.sh [hunk 2]
```diff
@@ -27,7 +28,10 @@ echo "1.create directory"
 # If install Path equal to "/" or related path is "/" or is empty, will cause directory "/bin" be overwrite or file adding,
 # so we should check its value. Here use command `realpath` to get the related path, and it will skip if your shell env
 # without command `realpath`.
-if [ ! -d $installPath ];then
+if [ ${baseDir} = $installPath ]; then
+  echo "Parameter installPath can not be same as the current path (pwd)."
```
Comment (DarkAssassinator) [inline script/install.sh]: echo "Fatal: The installPath can not be same as the current path: ${installPath}"  may be better, WDYT.

Comment (DarkAssassinator) [reply script/install.sh]: @EricGao888 @caishunfeng PTAL

Comment (muggleChen) [reply script/install.sh]: good point

**Comment by muggleChen** [comment]
Can you help review this pr ？ I removed the default codeowners by mistake. 
@caishunfeng @SbloodyS @zhongjiajie @EricGao888

**Comment by muggleChen** [comment]
Can you help review this pr? I removed the default reviewers by mistake.
@caishunfeng @SbloodyS @zhongjiajie @EricGao888

**Review by SbloodyS — APPROVED** [review]
LGTM. Thanks.

**Comment by muggleChen** [comment]
hi @EricGao888 , Could you please help merge this ,thanks

**Review by EricGao888 — APPROVED** [review]

**Pull request merged**